src/js/express.js

```diff
@@ -0,0 +1,179 @@
+
+// Look up the path (parent, sibling, child) relationships for a node.
+// Used in constructing the hierarchy we will use to wire together everything in our onCreate method.
+//
+// This does not require an invoker as it is basically a static method.
+gpii.express.pathForComponent = function (that) {
+    var instantiator  = fluid.getInstantiator(that);
+    var path          = instantiator.idToPath(that.id);
+    var segments      = fluid.model.parseEL(path);
+    return segments;
+};
+
+// Make a note of the lineage of this middleware or router component, all the way back presumably to this gpii.express
+// instance itself.
+//
+// This information will be used to wire together all nested routers and any middleware dependencies they have.
+gpii.express.registerComponentLineage = function(childComponent, expressComponent) {
+    // Get the list of parents for this component
+    var segments = gpii.express.pathForComponent(childComponent);
+
+    var childNickname  = segments[segments.length - 1];
+
+    var expressSegments = gpii.express.pathForComponent(expressComponent);
+
+    // If we only have one segment more than express, we are a child of express itself.
+    // We need this to be able to handle cases in which express itself is a child component.
+    if (segments.length === expressSegments.length + 1) {
+        expressComponent.options.members.directChildrenOfInterest.push(childNickname);
+    }
+    // Otherwise, register the nickname of this component and its immediate parent path, each level will take care of the next highest.
+    else if (segments.length > expressSegments.length + 1) {
+        var parentPath = segments.slice(expressSegments.length, segments.length - 1).join(".");
+        if (expressComponent.options.members.childrenByParent[parentPath]) {
+            expressComponent.options.members.childrenByParent[parentPath].push(childNickname);
+        }
+        else {
+            expressComponent.options.members.childrenByParent[parentPath] = [childNickname];
+        }
+    }
+};
+
+// Wire a child to its immediate descendants.
+gpii.express.connectDirectDescendants = function(that, childComponent, childPath) {
+    var descendants =  that.options.members.childrenByParent[childPath];
+
+    // This component has descendants, wire them in first
+    if (descendants !== undefined) {
+        // Wire together one level of routers and middleware, then recurse
+        for (var a = 0; a < descendants.length; a++) {
+            var grandchildNickname  = descendants[a];
+            var grandchildComponent = childComponent[grandchildNickname];
+            if (fluid.hasGrade(grandchildComponent.options, "gpii.express.router")) {
+                // We have to wire our children in with our path to preserve relative pathing, so that their router will begin with our path.
+                childComponent.options.router.use(childComponent.options.path, grandchildComponent.options.router);
```

Comment:
Condense reeferences to childComponent.options in this function. Also I think you should remove one layer of "parentage" by simply calling the incoming components "component" and "path" which will allow you to then map grandchild -> child, requiring less mental effort in readers...